parts/linux/cloud-init/artifacts/ensure_imds_restriction.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+# Ensure IMDS restriction rule is correctly set on the node, this script will be registered as kubelet service's ExecStart
+# to ensure desired iptables rule will be ensured after the node is restarted.
+# Below 3 variables are passed from kubelet service's drop-in file.
+primaryNicIP="${PRIMARY_NIC_IP}"
+insertRuleToMangleTable="${INSERT_IMDS_RESTRICTION_RULE_TO_MANGLE_TABLE}"
+enableIMDSRestriction="${ENABLE_IMDS_RESTRICTION}"
+
+ensureIMDSRestrictionRule() {
```

Comment:
you'd need to duplicate it within the script itself, but I'd probably recommend using `logs_to_events` like we currently do currently when configuring the restriction rule: https://github.com/Azure/AgentBaker/blob/0515c7aee97fc172541691a715ce779ca87cb8e5/parts/linux/cloud-init/artifacts/cse_helpers.sh#L389

this way we'll still get corresponding event telemetry

Reply:
updated the script here to call `logs_to_events`, curious will `logs_to_events` function be available here?

Reply:
performed an E2E test, `logs_to_events` is not available in kubelet's execprestart script, so update the code back.

Reply:
sorry I should've been more clear, I meant that you would have to duplicate the implementation of `logs_to_events` within the script directly in order to use it

on second thought though, I think we'd mainly want event telemetry around the lookup against IMDS, which is still happening via CSE scripts, so I think leaving as is would be fine

Reply:
got your point, let's follow-up in below conversation.